### PR TITLE
Update Beefy TS to Quarkus 2.x (Vertx 4 - 999-SNAPSHOT)

### PR DESCRIPTION
**Pull request by Sgitario**
Changes:
- Change HttpMethod to use the one from vertx route:
The reactive routes signature have changed and need to use another HttpMethod enum. More info in here: https://github.com/quarkusio/quarkus/commit/72da8546b564aa4871ea16aa4bf5b930f0f9286b

- Regression issue: Infinispan and Kafka extensions stopped working at the same time, reported by https://github.com/quarkusio/quarkus/issues/16386

- The annotation `@Health` got removed. I changed it to `@Readiness`. I've also added a new test case for covering the `HealthGroup` annotation and reported https://github.com/quarkusio/quarkus/issues/16389

- Regression Test: Validation errors on Reactive routes are no longer propagated. Reported by https://github.com/quarkusio/quarkus/issues/16430

- Regression issue: SmallRye OpenAPI generates incorrect content type when using Spring Web annotations. Reported by https://github.com/quarkusio/quarkus/issues/16395

### CODE REVIEW

File: 101-javaee-like-getting-started/src/main/java/io/quarkus/qe/core/GreetingHealthCheck.java [hunk 1]
```diff
@@ -1,12 +1,13 @@
 package io.quarkus.qe.core;
 
-import org.eclipse.microprofile.health.Health;
```
Comment (rsvoboda) [inline 101-javaee-like-getting-started/src/main/java/io/quarkus/qe/core/GreetingHealthCheck.java:3]: `Health` annotation got removed in Micro Profile 4 ?

File: 101-javaee-like-getting-started/src/main/java/io/quarkus/qe/core/GreetingHealthCheck.java [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
+import javax.enterprise.context.ApplicationScoped;
+
 import org.eclipse.microprofile.health.HealthCheck;
 import org.eclipse.microprofile.health.HealthCheckResponse;
 
-import javax.enterprise.context.ApplicationScoped;
+import io.smallrye.health.api.HealthGroup;
 
-@Health
+@HealthGroup("greeting")
```
Comment (rsvoboda) [inline 101-javaee-like-getting-started/src/main/java/io/quarkus/qe/core/GreetingHealthCheck.java:10]: Why SRye specific annotation and not the MicroProfile one?

File: 101-javaee-like-getting-started/src/test/java/io/quarkus/qe/core/JavaEELikeHealthCheckTest.java [hunk 3]
```diff
@@ -16,12 +22,29 @@ public void testHealthEndpoint() {
                 .then()
                 .statusCode(200)
                 .body("status", is("UP"),
-                        "checks.name", containsInAnyOrder("liveness", "readiness", "greeting", "Database connections health check"),
```
Comment (rsvoboda) [inline 101-javaee-like-getting-started/src/test/java/io/quarkus/qe/core/JavaEELikeHealthCheckTest.java:19]: I would like to still see the dummy `greeting` health check

Comment (rsvoboda) [reply 101-javaee-like-getting-started/src/test/java/io/quarkus/qe/core/JavaEELikeHealthCheckTest.java:19]: Or you intentionally repurposed the `greeting` check for newer stuff?

File: 101-javaee-like-getting-started/src/main/java/io/quarkus/qe/core/GreetingHealthCheck.java (hunk 1 above)
Comment (Sgitario) [reply 101-javaee-like-getting-started/src/main/java/io/quarkus/qe/core/GreetingHealthCheck.java:3]: The annotation `@Health` was deprecated in version 2.2 (Quarkus 1.13.1.Final) and finally dropped in version 3 (Quarkus 2.x).
In order to inject a custom health check in liveness and/or readiness, you need to use the annotations `@Liveness` and `@Readiness`.

File: 101-javaee-like-getting-started/src/main/java/io/quarkus/qe/core/GreetingHealthCheck.java (hunk 2 above)
Comment (Sgitario) [reply 101-javaee-like-getting-started/src/main/java/io/quarkus/qe/core/GreetingHealthCheck.java:10]: There is no something similar in MicroProfile, but in SRye. 
Another option is to cover the annotations @Liveness and @Readiness, but this functionality is well covered in upstream already. 
What do you think?

File: 101-javaee-like-getting-started/src/test/java/io/quarkus/qe/core/JavaEELikeHealthCheckTest.java (hunk 3 above)
Comment (Sgitario) [reply 101-javaee-like-getting-started/src/test/java/io/quarkus/qe/core/JavaEELikeHealthCheckTest.java:19]: I only thought it was a good idea to cover the HealthGroups. But I can revert this test case and change the old `@Health` by `@Liveness` if you prefer so. 
What do you think?

File: 101-javaee-like-getting-started/src/main/java/io/quarkus/qe/core/GreetingHealthCheck.java (hunk 1 above)
Comment (Sgitario) [reply 101-javaee-like-getting-started/src/main/java/io/quarkus/qe/core/GreetingHealthCheck.java:3]: I've changed the @Health to @Readiness and added a new test case to cover @HealthGroup.

File: 101-javaee-like-getting-started/src/main/java/io/quarkus/qe/core/GreetingHealthCheck.java (hunk 2 above)
Comment (Sgitario) [reply 101-javaee-like-getting-started/src/main/java/io/quarkus/qe/core/GreetingHealthCheck.java]: I've changed the `@Health` to `@Readiness` and added a new test case to cover `@HealthGroup`.

File: 101-javaee-like-getting-started/src/test/java/io/quarkus/qe/core/JavaEELikeHealthCheckTest.java (hunk 3 above)
Comment (Sgitario) [reply 101-javaee-like-getting-started/src/test/java/io/quarkus/qe/core/JavaEELikeHealthCheckTest.java]: I've changed the `@Health` to `@Readiness` and added a new test case to cover `@HealthGroup`.

**Comment by Sgitario** [comment]
Still failing because of this test:

```
Error:  Tests run: 3, Failures: 1, Errors: 0, Skipped: 2, Time elapsed: 106.966 s <<< FAILURE! - in io.quarkus.qe.vertx.sql.dbpool.PostgresPoolTest
Error:  checkIdleExpirationTime  Time elapsed: 95.491 s  <<< FAILURE!
```

@pjgg is helping to find out the root cause.

**Comment by Sgitario** [comment]
@rsvoboda this PR is ready to be merged. The test suite should be passing now.

**Review by pjgg — APPROVED** [review]

**Review by rsvoboda — APPROVED** [review]

**Pull request merged**